drizzlepac/haputils/poller_utils.py

```diff
@@ -634,35 +664,61 @@ def parse_obset_tree(det_tree, log_level):
-            log.debug("Attach the filter object {} to its associated total detection product object {}/{}.".format(filt_obj.filters,
-                                                                                                                   tdp_obj.instrument,
-                                                                                                                   tdp_obj.detector))
-            # Identify what exposures should use single-image CR identification algorithm
-            is_ccd = not (filt_obj.instrument.lower() == 'wfc3' and filt_obj.detector.lower() == 'ir')
-            if is_ccd and len(filt_obj.edp_list) == 1:
-                for e in filt_obj.edp_list:
-                    e.crclean = True
+                    # Populate total detection product dictionary with input filename
+                    obset_products[totprod]['files'].append(filename[1])
 
-            tdp_obj.add_product(filt_obj)
+                    # Append filter object to the list of filter objects for this specific total product object
+                    log.debug("Attach the filter object {} to its associated total detection product object {}/{}.".format(filt_obj.filters,
+                                                                                                                           tdp_obj.instrument,
+                                                                                                                           tdp_obj.detector))
+                    # Identify what exposures should use single-image CR identification algorithm
+                    is_ccd = not (filt_obj.instrument.lower() == 'wfc3' and filt_obj.detector.lower() == 'ir')
+                    if is_ccd and len(filt_obj.edp_list) == 1:
+                        for e in filt_obj.edp_list:
+                            e.crclean = True
+
+                    #tdp_obj.add_product(filt_obj)
+
+                elif is_grism:
+                    tdp_obj.add_grism_member(grism_sep_obj)
+
+            if not is_grism:
+                tdp_obj.add_product(filt_obj)
 
         # Add the total product object to the list of TotalProducts
         tdp_list.append(tdp_obj)
 
+    # If the total product object has Grism members, it MUST have direct exposure
+    # members too, or the total product object is not valid.  In this case,
+    # the total product object, as well as its object contents must be deleted.
+    # Also, the SVM FLT/FLC files for the Grism/Prism images must be deleted.
+    index_to_delete = []
+    for index, tdp in enumerate(tdp_list):
+        if tdp.grism_edp_list and not tdp.edp_list:
+            for item in tdp.grism_edp_list:
+                try:
+                    os.remove(item.full_filename)
+                except OSError:
+                    pass
+            index_to_delete.append(index)
+            tdp.grism_edp_list.clear()
+            del tdp.grism_edp_list[:]
+    for item in index_to_delete:
+        del tdp_list[item]
```

Comment:
Whenever you are deleting multiple items from a list, it should be done in reverse order (last items deleted first).  Think about `a = ['x', 'y', 'z', '0', '1', '2']` and you want to delete 'z' and '1' at indices [2, 4].  If you do `for item in [2,4]: del a[item]', you will end up with `a = ['x','y','0','1']`, but if you do 'for item in [4,2]: del a[item]', you will get what you want.  I am afraid the deletion in 713 and 714 will run into problems with this problem.  Please check whether you need to add `index_to_delete.reverse()` before line 713.

Reply:
Indeed.  I was trying to be careful about this very issue, but worried more about deletion of the Grism list attribute in the object than the Total object from Total product list!  Worked correctly in testing as the list only had two items.  Fixed in git commit 1fce265.